waiter/integration/waiter/basic_test.clj

```diff
@@ -699,7 +705,31 @@
                       :interval 2 :timeout 30))
                   (keys router-id->router-url))]
             (is (every? #(true? (val %)) router-id->service-id-deleted)
-                (str service-id " present in at least one router: " router-id->service-id-deleted))))))))
+                (str service-id " present in at least one router: " router-id->service-id-deleted))))
+
+        (let [{:keys [service-id cookies]}
+              (make-request-with-debug-info {:x-waiter-name (rand-name)} #(make-kitchen-request waiter-url %))]
+          (testing "service-known-on-all-routers"
+            (assert-service-on-all-routers waiter-url service-id cookies))
+
+          (testing "delete service successfully with timeout of 10000ms"
+            (let [{:keys [body] :as response} (make-request waiter-url (str "/apps/" service-id) :method :delete :query-params "timeout=10000")]
+              (assert-response-status response http-200-ok)
+              (is (get (json/read-str body) "routers-agree"))))
+
+          (testing "service-deleted-from-all-routers"
+            (let [router-id->service-id-deleted
+                  (pc/map-from-keys
+                    (fn [router-id]
+                      (let [router-url (router-id->router-url router-id)
+                            {:keys [body]} (make-request router-url "/apps" :cookies cookies)]
+                        (->> (try-parse-json (str body))
+                             (filter #(= service-id (get % "service-id")))
+                             seq
+                             not)))
+                    (keys router-id->router-url))]
```

Comment:
Nitpick: this is more diomatic:
```
(doseq [[router-id router-url] (seq router-id->router-url]
   ...)
```

Reply:
doseq returns nil; how would I use it properly in this situation along with the assertion on the next line
`(is (every? #(true? (val %)) router-id->service-id-deleted)`

Reply:
My bad with `doseq`, `for-map` may be a better example, but this is a nitpick, so feel free to ignore.
Example:   (let [token->current-metadata (pc/for-map
                                  [{:strs [etag last-update-time owner token]} (load-token-list cluster-url)]
                                  token
                                  {:etag etag
                                   :last-update-time (some-> last-update-time utils/iso8601->millis)
                                   :owner owner})